weave-js/src/core/ops/domain/artifactVersion.ts

```diff
@@ -114,6 +114,22 @@ export const opArtifactVersionSize = makeArtifactVersionOp({
   resolver: ({artifactVersion}) => artifactVersion.size,
 });
 
+export const opArtifactVersionTTLDurationSeconds = makeArtifactVersionOp({
+  name: 'artifactVersion-ttlDurationSeconds',
+  argTypes: artifactVersionArgTypes,
+  description: `Returns the ttl duration in seconds that ${docType(
+    'artifactVersion'
+  )} takes to mark as deleted`,
+  argDescriptions: {
+    artifactVersion: artifactVersionArgDescription,
+  },
+  returnValueDescription: `The ttl duration in seconds for ${docType(
+    'artifactVersion'
+  )}`,
+  returnType: inputTypes => 'number',
+  resolver: ({artifactVersion}) => artifactVersion.ttlDurationSeconds,
```

Comment:
is this guaranteed to be set for all artifact versions?

Reply:
yes there is always an int, is there more work that needs to be done for backwards compatibility though? In the frontend code itself I'm still checking for the case where ttl is undefined just in case